src/contract_wrappers/modules/sto/usd_tiered_sto_wrapper/__tests__/3.1.0.test.ts

```diff
@@ -265,35 +331,185 @@ describe('USD Tiered STO 3.1.0', () => {
+      verify(mockedSecurityTokenOwnerMethod.callAsync()).once();
+      verify(mockedSecurityTokenContract.owner).once();
+      verify(mockedWrapper.getAvailableAddressesAsync()).once();
+      // denominatedCurrency
+      verify(mockedContract.denominatedCurrency).once();
+      verify(mockedDenominatedCurrencyMethod.callAsync()).once();
+      // startTime
+      verify(mockedContract.startTime).once();
+      verify(mockedStartTimeMethod.callAsync()).once();
+      // fundRaiseTypes
+      verify(mockedContract.fundRaiseTypes).twice();
+      verify(mockedFundRaiseTypesPolyMethod.callAsync(FundRaiseType.ETH)).once();
+      verify(mockedContract.modifyOracles).once();
+      verify(
+        mockedMethod.sendTransactionAsync(
+          mockedParams.customOracleAddresses,
+          objectContaining(stringToBytes32(mockedParams.denominatedCurrencySymbol)),
+          mockedParams.txData,
+          mockedParams.safetyFactor,
+        ),
+      ).once();
+    });
+
+    test('should modify oracles without oracle addresses', async () => {
+      // Owner Address expected
+      const expectedOwnerResult = '0x5555555555555555555555555555555555555555';
+      // Security Token Address expected
+      const expectedSecurityTokenAddress = '0x3333333333333333333333333333333333333333';
+      // Setup get Security Token Address
+      const mockedGetSecurityTokenAddressMethod = mock(MockedCallMethod);
+      when(mockedContract.securityToken).thenReturn(instance(mockedGetSecurityTokenAddressMethod));
+      when(mockedGetSecurityTokenAddressMethod.callAsync()).thenResolve(expectedSecurityTokenAddress);
+
+      when(mockedContractFactory.getSecurityTokenContract(expectedSecurityTokenAddress)).thenResolve(
+        instance(mockedSecurityTokenContract),
       );
+      const mockedSecurityTokenOwnerMethod = mock(MockedCallMethod);
+      when(mockedSecurityTokenOwnerMethod.callAsync()).thenResolve(expectedOwnerResult);
+      when(mockedSecurityTokenContract.owner).thenReturn(instance(mockedSecurityTokenOwnerMethod));
+
+      // Mock web3 wrapper owner
+      when(mockedWrapper.getAvailableAddressesAsync()).thenResolve([expectedOwnerResult]);
+
+      const mockedParams = {
+        customOracleAddresses: [],
+        denominatedCurrencySymbol: '',
+        txData: {},
+        safetyFactor: 10,
+      };
+      const expectedResult = getMockedPolyResponse();
+      // Mocked method
+      const mockedMethod = mock(MockedSendMethod);
+
+      // denominatedCurrency mock
+      const expectedDenominatedCurrencyResult = stringToBytes32('USDT');
+      const mockedDenominatedCurrencyMethod = mock(MockedCallMethod);
+      when(mockedContract.denominatedCurrency).thenReturn(instance(mockedDenominatedCurrencyMethod));
+      when(mockedDenominatedCurrencyMethod.callAsync()).thenResolve(expectedDenominatedCurrencyResult);
+
+      // startTime mock
+      const expectedStartTimeResult = new BigNumber(3876536279);
```

Comment:
Maybe we could convert a real date to big number, I’m not sure if the precise date matters here or not